hudi-flink-datasource/hudi-flink/src/test/java/org/apache/hudi/sink/TestWriteCopyOnWrite.java

```diff
@@ -538,11 +539,20 @@ public void testWriteMultiWriterInvolved() throws Exception {
         .assertNextEvent()
         .checkpointComplete(1)
         .checkWrittenData(EXPECTED3, 1);
-    // step to commit the 2nd txn, should throw exception
-    // for concurrent modification of same fileGroups
-    pipeline1.checkpoint(1)
-        .assertNextEvent()
-        .checkpointCompleteThrows(1, HoodieWriteConflictException.class, "Cannot resolve conflicts");
+    // step to commit the 2nd txn
+    pipeline1 = pipeline1
+        .checkpoint(1)
+        .assertNextEvent();
+    if (OptionsResolver.isLocklessMultiWriter(conf)) {
+      // should success for concurrent modification of same fileGroups if using lockless multi writers
```

Comment:
It should be always true? `OptionsResolver.isLocklessMultiWriter(conf)`

Reply:
No, it's only be true if all the following condition is satisfied:
* the table is MOR table
* the index is simple bucket index
* enable OPTIMISTIC_CONCURRENCY_CONTROL

So it could be false for COW table or MOR table with other index type.